src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -271,6 +272,48 @@ private string GetAnalyzerPath(AnalyzerReference analyzerReference)
             return analyzerReference.FullPath;
         }
 
+        protected override void ApplyParseOptionsChanged(ProjectId projectId, ParseOptions options)
+        {
+            if (projectId == null)
+            {
+                throw new ArgumentNullException(nameof(projectId));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            var parseOptionsService = CurrentSolution.GetProject(projectId).LanguageServices.GetService<IParseOptionsService>();
+            if (parseOptionsService == null)
+            {
+                return;
+            }
+
+            string newVersion = parseOptionsService.GetLanguageVersion(options);
+
+            GetProjectData(projectId, out var hostProject, out var hierarchy, out var project);
+            foreach (string configurationName in (object[])project.ConfigurationManager.ConfigurationRowNames)
+            {
+                switch (hostProject.Language)
+                {
+                    case LanguageNames.CSharp:
+                        var csharpProperties = (VSLangProj80.CSharpProjectConfigurationProperties3)project.ConfigurationManager
+                            .ConfigurationRow(configurationName).Item(1).Object;
+
+                        if (parseOptionsService.NewerThan(newVersion, csharpProperties.LanguageVersion))
```

Comment:
This should be !=, since people might do an apply changes to move to a lower version either.